tests/test_cpu_throttle/assert

```diff
@@ -8,92 +8,99 @@ import os
 import sys
 import time
 from glob import glob
+import coloredlogs
+coloredlogs.install(level='INFO')
 
 class TestCPUThrottle:
     def setup_class(self):
-        self.test_flag = True
         print("CPU throttle test")
-        sh.mkdir('/sys/fs/cgroup/cpu/test')
-        
+        self.cgpath = '/sys/fs/cgroup/cpu/test'
+        sh.tee(sh.echo(1), '/sys/fs/cgroup/cpu/cgroup.clone_children')
+        sh.mkdir(self.cgpath)
+
     def init_cgroup(self):
-        cmd = "while :; do :; done"
-        self.start_time = time.time()
-        self.child = subprocess.Popen(cmd, shell=True)
-        sh.echo(self.child.pid, _out='/sys/fs/cgroup/cpu/test/cgroup.procs')
+        self.child = sh.bash('-c', 'while :; do :; done', _bg=True, bg_exc=False)
+        sh.tee(sh.echo(self.child.pid), '%s/cgroup.procs' % self.cgpath)
 
     def set_cfs_quota(self, t_us):
-        sh.echo(t_us, _out='/sys/fs/cgroup/cpu/test/cpu.cfs_quota_us')
-        time.sleep(10)
+        sh.echo(t_us, _out='%s/cpu.cfs_quota_us' % self.cgpath)
 
     def test_all(self):
-        self.set_cfs_quota('50000')
+        self.install_module()
         self.init_cgroup()
+        self.set_cfs_quota('50000')
         self.check_le_75()
         self.check_after_load()
         self.set_cfs_quota('100000')
         self.check_gt_75()
         self.check_after_unload()
-        
+
     def check_le_75(self):
         cpu_util = self.get_cpu_util(self.child.pid)
-        # assert cpu_util <= 75
-        if cpu_util > 75:
-            self.error_handler(0, 75)
+        self.validate_lt(cpu_util, 75)
 
-    def check_after_load(self):
+    def install_module(self):
         scheduler_rpm = glob(os.path.join('/tmp/work', 'scheduler*.rpm'))
         if len(scheduler_rpm) != 1:
             print("Please check your scheduler rpm");
             self.teardown_class()
             sys.exit(1)
         scheduler_rpm = scheduler_rpm[0]
         sh.rpm('-ivh', scheduler_rpm)
+        sh.tee(sh.echo(0), '/sys/kernel/plugsched/plugsched/enable')
+
+    def uninstall_module(self):
+        sh.rpm('-e', 'scheduler-xxx')
+
+    def check_after_load(self):
+        sh.tee(sh.echo(1), '/sys/kernel/plugsched/plugsched/enable')
         cpu_util = self.get_cpu_util(self.child.pid)
-        # assert cpu_util <= 75
-        if cpu_util > 75:
-            self.error_handler(0, 75)
+        self.validate_lt(cpu_util, 75)
 
     def check_gt_75(self):
         cpu_util = self.get_cpu_util(self.child.pid)
-        # assert cpu_util >= 75
         if cpu_util < 75:
             self.error_handler(1, 75)
```

Comment:
Is it better to use the self.validate_gt()? This member function has been defined.

Reply:
Okay